sled-agent/src/rack_setup/service.rs

```diff
@@ -1329,6 +1329,31 @@ pub(crate) fn build_initial_blueprint_from_sled_configs(
     sled_configs: BTreeMap<Uuid, SledConfig>,
     internal_dns_version: Generation,
 ) -> Blueprint {
+    // Helper to convert an `OmicronZoneConfig` into a `BlueprintZoneConfig`.
+    // This is separate primarily so rustfmt doesn't lose its mind.
+    let to_bp_zone_config = |z: crate::params::OmicronZoneConfig| {
+        // All initial zones are in-service.
+        let disposition = BlueprintZoneDisposition::InService;
+        match BlueprintZoneConfig::from_omicron_zone_config(
+            z.into(),
+            disposition,
+        ) {
+            Ok(config) => config,
+            Err(InvalidOmicronZoneType::ParseSocketAddr {
+                kind,
+                addr,
+                err,
+            }) => {
+                // This error is impossible, because our `z.into()` converts all
+                // `SocketAddr`s into strings, which are all therefore valid.
+                unreachable!(
+                    "RSS produced {kind} zone with invalid address \
+                    {addr}: {err}"
+                )
```

Comment:
Can this return an error instead? I'm worried that some other caller will come about and call this with untrusted data -- instead, the caller which knows it is trusted can panic if this errors out.

Reply:
Yeah, changed in [5b290f1](https://github.com/oxidecomputer/omicron/pull/5523/commits/5b290f12b8236f6dcd8fb61660a80845fd6eaff4). This was simpler than I expected because there are already other ways RSS could construct an invalid plan (and those return errors already); I think I was a little tunnel visioned on this progenitor socketaddr format annoyance.